continuous_integration/environment-mindeps-optional.yaml

```diff
@@ -0,0 +1,54 @@
+name: test-environment
+channels:
+  - conda-forge
+dependencies:
+  # required dependencies
+  - packaging=20.0
+  - python=3.8
+  - pyyaml=5.3.1
+  - click=7.0
+  - cloudpickle=1.1.1
+  - partd=1.2.0
+  - fsspec=2021.09.0  # required for arrow I/O support
```

Comment:
This is the earliest `fsspec` version with an Arrow implementation - not sure if we want to bump the core dependency or just specify that this would be needed to do I/O with Parquet?

Reply:
Bumping the minimum `fsspec` to `2021.09.0` is reasonable -- it's been out for a while at this point.

Reply:
Bumping the minimum `fsspec` to `2021.09.0` is reasonable -- it's been out for a while at this point.

Reply:
Would be good to update elsewhere throughout the codebase too (e.g. `setup.py`, other mindeps builds)